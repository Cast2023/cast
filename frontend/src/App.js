import {
  Container,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableRow,
  Paper,
} from "@mui/material"
import { BrowserRouter as Router, Routes, Route, Link } from "react-router-dom"
import React, { useState, useEffect } from "react"
import axios from "axios"

const Notes = ({ notes }) => (
  <div>
    <h2>Notes</h2>

    <TableContainer component={Paper}>
      <Table>
        <TableBody>
          {notes.map((note) => (
            <TableRow key={note.id}>
              <TableCell>
                <Link to={`/notes/${note.id}`}>{note.content}</Link>
              </TableCell>
              <TableCell>{note.name}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  </div>
)

const App = () => {
  const [content, setContent] = useState([])
  const [newContent, setNewContent] = useState("a new note...")

  useEffect(() => {
    axios.get("http://127.0.0.1:8000/api/backend ").then((response) => {
      setContent(response.data)
    })
  }, [])

  const result = content.map((item) => <li> {item.content} </li>)

  const addContent = (event) => {
    event.preventDefault()
    const contentObject = {
      content: newContent,
    }

    axios
      .post("http://127.0.0.1:8000/api/backend ", contentObject)
      .then((response) => {
        console.log("LOGGING: ", response)
      })
      .catch((error) => {
        console.log(error, contentObject)
      })
    const addContentObject = {
      content: newContent,
      id: content.length + 1,
    }

    setContent(content.concat(addContentObject))
    setNewContent("")
  }
  const handleContentChange = (event) => {
    setNewContent(event.target.value)
  }

  return (
    <Container>
      <div>
        <p>Hello world</p>

        <Notes notes={result} />
        <div>
          <form onSubmit={addContent}>
            <input value={newContent} onChange={handleContentChange} />
            <br />
            <input type="submit" value="OK" />
          </form>
        </div>
      </div>
    </Container>
  )
}

export default App
