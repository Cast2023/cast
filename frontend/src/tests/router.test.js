import React from "react"
import { render, cleanup, screen } from "@testing-library/react"
import App from "../App"

afterEach(cleanup)

it("should take a snapshot", () => {
  const { asFragment } = render(<App />)

  expect(asFragment(<App />)).toMatchSnapshot()
})

it("renders learn react link", () => {
  render(<App />)
  const linkElement = screen.getByText(/Home page/i)
  expect(linkElement).toBeInTheDocument()
})
