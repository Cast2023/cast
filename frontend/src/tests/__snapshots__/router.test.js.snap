// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div>
      <h1>
        Competency, Allocation and Skill tracker
      </h1>
    </div>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <button
            aria-label="menu"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1d6w9lk-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <a
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1y942vo-MuiButtonBase-root-MuiButton-root"
            href="/"
            id="home"
            tabindex="0"
          >
            home
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
          <a
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1y942vo-MuiButtonBase-root-MuiButton-root"
            href="/notes"
            tabindex="0"
          >
            notes
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
          <a
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1y942vo-MuiButtonBase-root-MuiButton-root"
            href="/profile"
            tabindex="0"
          >
            profile
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
          <a
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1y942vo-MuiButtonBase-root-MuiButton-root"
            href="/search"
            tabindex="0"
          >
            search
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </a>
        </div>
      </header>
    </div>
    <div>
      <h2>
        Home
      </h2>
      <div>
        This is the home page.
      </div>
      <br />
      <div>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam vestibulum tincidunt risus, a scelerisque ipsum lacinia sed. Ut sodales vulputate ex. Nam et interdum elit. Nulla ornare id augue tincidunt condimentum. Etiam auctor, ipsum vel dictum gravida, diam magna consectetur felis, sed eleifend erat nibh quis dui. Duis a tincidunt felis, ut iaculis est. Suspendisse tincidunt vel diam at euismod. Suspendisse luctus maximus arcu id faucibus. Quisque eget orci et augue lacinia sagittis id id orci.
        <br />
        <br />
        Nulla nulla dui, iaculis eu mi id, congue molestie metus. Sed tincidunt ligula id aliquam ultrices. Integer tempor cursus elit, sed aliquet odio aliquam nec. Nam eget arcu iaculis, semper orci a, iaculis arcu. Suspendisse sed iaculis ligula, ut imperdiet tortor. Vestibulum nec elit sit amet nunc bibendum venenatis. Curabitur ullamcorper placerat lorem, ut aliquam odio tristique quis. Aenean pulvinar vulputate tellus quis fringilla. Sed mattis posuere odio, et condimentum risus viverra at. Aliquam erat volutpat.
      </div>
    </div>
    <div>
      <i>
        Cast APP, OhTu-projekti 2023
      </i>
    </div>
  </div>
</DocumentFragment>
`;
